### DISCUSSION TITLE
어떤 기준으로 적절한 함수형 인터페이스를 사용해야 하는가?

### DISCUSSION THREAD

**Discussion by coalong** [Study-2-Modern-Java-In-Action/Study-2-Modern-Java-in-Action#37] [CHAPTER 3 람다 표현식 - Q&A] [answered]
### 어떤 기준으로 적절한 함수형 인터페이스를 사용해야 하는가?

p. 109 
> 대상 형식(target typing)이라는 특징 때문에 하나의 람다 표현식이더라도 호환되는 추상 메서드를 가진 다른 함수형 인터페이스로 사용될 수 있다.

**Comment by JoisFe** [comment, marked as answer]
- 일단 표준 함수형 인터페이스를 사용하는게 좋다고 생각하고
- 그 중에서도 좀 더 구체적인 의미를 내포? 하는 함수형 인터페이스를 사용하는게 좋다고 생각합니다. (제 생각일 뿐.. 의견들 얘기해주세요)
``` java
Comparator<Apple> c1 = (Apple a1, Apple a2) -> a1.getWeight().compareTo(a2.getWeight); // (1)
ToIntBiFunction<Apple> c1 = (Apple a1, Apple a2) -> a1.getWeight().compareTo(a2.getWeight); // (2)
```

- 여기서 사과 두개의 무게를 비교하고 있기 때문에 (1)을 사용하지 않을까 싶습니다.
